osf/models/user.py

```diff
@@ -750,10 +751,11 @@ def merge_user(self, user):
             for key, value in user.mailchimp_mailing_lists.items():
                 # subscribe to each list if either user was subscribed
                 subscription = value or self.mailchimp_mailing_lists.get(key)
-                signals.user_merged.send(self, list_name=key, subscription=subscription)
+                from website.profile.views import update_mailchimp_subscription
+                update_mailchimp_subscription(self, list_name=key, subscription=subscription)
 
                 # clear subscriptions for merged user
-                signals.user_merged.send(user, list_name=key, subscription=False, send_goodbye=False)
+                update_mailchimp_subscription(user, list_name=key, subscription=False, send_goodbye=False)
```

Comment:
I feel a signal is not used correctly here. Ideally we should have one signal per event. I think it was used here to make it post-commit/ and async, but I think that's happening anyway. Needs more research.